### ISSUE TITLE
Adopt support for JDK 21

### ISSUE THREAD

**Issue by fniephaus** [oracle/graal#6854]
### TL;DR

The next feature release of GraalVM will exclusively support JDK 21.

### Goals

As GraalVM is adopting the [JDK release process](https://www.graalvm.org/2022/openjdk-announcement/), GraalVM will exclusively support JDK 21 in the [September release](https://www.graalvm.org/release-calendar/). This means users can take advantage of [all new features introduced with JDK 21](https://openjdk.org/projects/jdk/21/) with the Graal JIT compiler. All preview features and [generational ZGC](https://openjdk.org/jeps/439) will not be supported in Native Image. [Virtual threads](https://openjdk.org/jeps/444) preview support was available in Native Image since [JDK 19](https://medium.com/graalvm/graalvm-22-3-is-here-jdk-19-builds-jlink-support-new-monitoring-features-and-more-f6e2b2eeff95) and will be available as a finalized feature in JDK 21.

The community and end-users are encouraged to upgrade to JDK 21, but have several options if they need to stay on older JDK versions: The community edition of GraalVM for JDK 17 will see [another CPU update in October](https://www.graalvm.org/release-calendar/), while Oracle GraalVM for JDK 17 is [supported until 2025](https://docs.oracle.com/en/graalvm/jdk/17/docs/release-calendar/) with security updates and important bugfixes.

### Non-Goals

- Add Native Image support for all preview features

### Related PRs

- https://github.com/oracle/graal/pull/4802
- https://github.com/oracle/graal/pull/6835

**Comment by thomaswue** [comment]
We would prefer if other distributions release 17 from the same branch as the Community Edition. If there are important changes on master, they should be backported. We are taking this approach also for the Oracle GraalVM for Java 17 distribution.
